src/index.js

```diff
@@ -18,9 +19,11 @@ ReactDOM.render(
       <AuthProvider>
         <ToastProvider>
           <FilterProvider>
-            <VideoListProvider>
-              <App />
-            </VideoListProvider>
+            <UserDataProvider>
+              <VideoListProvider>
+                <App />
+              </VideoListProvider>
+            </UserDataProvider>
           </FilterProvider>
         </ToastProvider>
       </AuthProvider>

```

Comment:
Suggestion: You can create a separate component and club all contexts there and export just one Context which you can use here.

Reply:
Done..